### PR TITLE
Feature/pay run insights

### PR DESCRIPTION
**Pull request by dunnudge**
## Link to CU ticket

[Link](https://app.clickup.com/t/1px2g5z)

## Describe this PR

Implementation to get single pay run insights

### CODE REVIEW

File: LBH.AdultSocialCare.Api/V1/Gateways/Payments/Concrete/PayRunInvoiceGateway.cs [hunk 1]
```diff
@@ -89,6 +90,43 @@ public PayRunInvoiceGateway(DatabaseContext dbContext)
             return await BuildPayRunInvoiceQuery(query, fields).SingleOrDefaultAsync();
         }
 
+        public async Task<decimal> GetPayRunInvoicedTotalAsync(Guid payRunId)
```
Comment (brkzkn) [inline LBH.AdultSocialCare.Api/V1/Gateways/Payments/Concrete/PayRunInvoiceGateway.cs:93]: No need to go DB for each metric. 

Let's get PayrunInvoices with related tables then calculate all metrics.

File: LBH.AdultSocialCare.Api/V1/UseCase/Payments/Concrete/GetPayRunInsightsUseCase.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        private readonly IPayRunInvoiceGateway _payRunInvoiceGateway;
+
+        public GetPayRunInsightsUseCase(IPayRunGateway payRunGateway, IPayRunInvoiceGateway payRunInvoiceGateway)
+        {
+            _payRunGateway = payRunGateway;
+            _payRunInvoiceGateway = payRunInvoiceGateway;
+        }
+
+        public async Task<PayRunInsightsResponse> GetAsync(Guid payRunId)
+        {
```
Comment (brkzkn) [inline LBH.AdultSocialCare.Api/V1/UseCase/Payments/Concrete/GetPayRunInsightsUseCase.cs:22]: You have already payrun in here, no need to go db for each metrics

**Review by brkzkn — CHANGES_REQUESTED** [review]

File: LBH.AdultSocialCare.Api/V1/UseCase/Payments/Concrete/GetPayRunInsightsUseCase.cs (hunk 2 above)
Comment (dunnudge) [reply LBH.AdultSocialCare.Api/V1/UseCase/Payments/Concrete/GetPayRunInsightsUseCase.cs:22]: The pay run here has no detail. No navigation property included

Comment (brkzkn) [reply LBH.AdultSocialCare.Api/V1/UseCase/Payments/Concrete/GetPayRunInsightsUseCase.cs:22]: That's right. You can get payrun with child tables and no need to go db ;) 

query for each time is more costly than getting big data volume.

**Review by brkzkn — APPROVED** [review]

**Pull request merged**